lib/upload_creator.rb

```diff
@@ -117,12 +127,16 @@ def create_for(user_id)
       @upload.for_export          = true if @opts[:for_export]
       @upload.for_site_setting    = true if @opts[:for_site_setting]
       @upload.for_gravatar        = true if @opts[:for_gravatar]
+      uploading_in_public_context = @upload.for_theme || @upload.for_site_setting || @opts[:type] == "avatar"
+
+      # determine whether upload should be secure when first created
```

Comment:
Right now there are two places where we determine whether an upload should be marked secure: here and in `models/upload.rb#should_be_secure?`. I'm wondering whether it would make sense to abstract the logic into one central place. There are some differences (due to the upload existing in one case and not the other) but there is a fair amount of overlap, too.

Reply:
@pmusaraj very good point, I have moved this stuff out into a new `UploadSecurity` class :+1: